jsx/Form.js

```diff
@@ -411,6 +411,271 @@ var SelectElement = React.createClass({
+        {Object.keys(options).map(function(option) {
+          return (
+            <option value={option} key={option}>{options[option]}</option>
+          );
+        })}
+      </select>;
+      // else, use a text input by default
+    } else {
+      input = <input
+        type="text"
+        name={this.props.name}
+        id={this.props.id}
+        className="form-control"
+        value={this.props.value || ""}
+        disabled={disabled}
+        onChange={this.handleChange}
+        onKeyPress={this.handleKeyPress}
+      />;
+    }
+
+    // iterate through added Tags items and render them
+    // with deletion button
+    let items = this.props.items.map(function(item) {
+      let itmTxt;
+      // in event that the passed item is a key of options,
+      // render option value
+      if (Object.keys(options).length > 0 && options[item] !== undefined) {
+        itmTxt = options[item];
+        // otherwise just render item as is
+      } else {
+        itmTxt = item;
+      }
+      return (
+          <button
+            className="btn btn-info btn-inline"
+            type="button"
+            onClick={this.handleRemove}
+            data-item={item}
+          >
+            {itmTxt}
+            &nbsp;
+            <span
+              className="glyphicon glyphicon-remove"
+              data-item={item}
+            />
+          </button>
+      );
+    }, this);
+    return (
+      <div className={elementClass}>
+        <label className="col-sm-3 control-label" htmlFor={this.props.id}>
+          {this.props.label}
+          {requiredHTML}
+        </label>
+        <div className="col-sm-9">
+          {items}
+          {input}
+          {errorMessage}
+          <button
+            className="btn btn-success btn-add-tag"
+            id={this.props.id + 'Add'}
+            type="button"
+            onClick={this.handleAdd}
+            >
+            <span className="glyphicon glyphicon-plus"/>
+            {this.props.btnLabel}
+          </button>
+        </div>
+      </div>
+    );
+  }
+}
+
+TagsElement.propTypes = {
+  name: React.PropTypes.string.isRequired,
+  id: React.PropTypes.string.isRequired,
+  pendingValKey: React.PropTypes.string.isRequired,
+  options: React.PropTypes.object,
+  items: React.PropTypes.array,
+  label: React.PropTypes.string,
+  value: React.PropTypes.string,
+  class: React.PropTypes.string,
+  multiple: React.PropTypes.bool,
+  required: React.PropTypes.bool,
+  disabled: React.PropTypes.bool,
+  emptyOption: React.PropTypes.bool,
+  errorMessage: React.PropTypes.string,
+  btnLabel: React.PropTypes.string,
+  allowDupl: React.PropTypes.bool,
+  useSearch: React.PropTypes.bool,
+  strictSearch: React.PropTypes.bool,
+  onUserInput: React.PropTypes.func,
+  onUserAdd: React.PropTypes.func,
+  onUserRemove: React.PropTypes.func
+};
+
+TagsElement.defaultProps = {
+  name: '',
+  options: {},
+  items: [],
+  label: '',
+  value: undefined,
+  id: '',
```

Comment:
`id` should not default to empty string as this makes the forms prone to `id` value duplication.

Reply:
i agree! maybe a separate small PR can do this to keep the refactor PR within scope?

Reply:
changes made here instead https://github.com/aces/Loris/pull/3597/commits/a2c2aa1d34908d36064b42071d1d377a67cc5eab